go/client/cmd_wait.go

```diff
@@ -0,0 +1,92 @@
+package client
+
+import (
+	"fmt"
+	"github.com/keybase/cli"
+	"github.com/keybase/client/go/libcmdline"
+	"github.com/keybase/client/go/libkb"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/go-framed-msgpack-rpc/rpc"
+	"golang.org/x/net/context"
+	"time"
+)
+
+type CmdWait struct {
+	libkb.Contextified
+	duration int
+}
+
+const maxWaitTime = 60
+
+func NewCmdWait(cl *libcmdline.CommandLine, g *libkb.GlobalContext) cli.Command {
+	return cli.Command{
+		Name:  "wait",
+		Usage: "Waits for the keybase server to start up",
+		Action: func(c *cli.Context) {
+			cl.SetForkCmd(libcmdline.NoFork)
+			cl.SetLogForward(libcmdline.LogForwardNone)
+			cl.ChooseCommand(&CmdWait{Contextified: libkb.NewContextified(g)}, "wait", c)
+		},
+		Flags: []cli.Flag{
+			cli.IntFlag{
+				Name:   "duration, d",
+				Value: 10,
+				Usage:  "How many seconds to wait before timing out",
+			},
+		},
+	}
+}
+
+func (c *CmdWait) ParseArgv(ctx *cli.Context) error {
+	if len(ctx.Args()) > 1 {
+		return UnexpectedArgsError("wait")
+	}
+
+	c.duration = ctx.Int("duration")
+	if c.duration <= 0 || c.duration > maxWaitTime {
+		return fmt.Errorf("invalid duration %d, must be between 1 and %d", c.duration, maxWaitTime)
+	}
+	return nil
+}
+
+func (c *CmdWait) Run() error {
+	for i := 0; i < c.duration; i++ {
```

Comment:
This doesn't quite work in case GetFullStatus takes a non negligible amount of time. Better to use a context with a timeout on it: https://golang.org/pkg/context/#WithTimeout.

Reply:
I don't need to worry about this with the reduced status check right?